app/src/main/java/com/example/taskthree/view/CounterFragment.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.taskthree.view
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import com.example.taskthree.databinding.FragmentCounterBinding
+import com.example.taskthree.viewmodel.CounterViewModel
+
+class CounterFragment : Fragment() {
+    private var counter: Int = 0
+    private lateinit var binding: FragmentCounterBinding
+    private val viewModel: CounterViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        binding = FragmentCounterBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.textviewCounter.text = counter.toString()
+        binding.buttonIncrease.setOnClickListener {
+            if (binding.switchButton.isChecked) {
+                viewModel.incrementCounter()
+            } else {
+                increaseCounter()
+            }
+        }
+
+        binding.switchButton.setOnCheckedChangeListener { _, isChecked ->
+            if (isChecked) {
+                binding.textviewCounter.text = viewModel.counterObserver.value.toString()
+                viewModel.counterObserver.observe(viewLifecycleOwner) { counter ->
```

Comment:
Bu observer'in switch listener icerisinde olmasina gerek yok. isChecked true olmasa dahi counter'i dinlememizde sakinca yok gozukuyor

Reply:
abi switch listenerın içinden çıkarınca observer'ı, switch butonu on off yaptığımda textviewCounter senkron olarak değişmiyor. Ondan içinden bıraktım